### DISCUSSION TITLE
Adding more custom command placeholders

### DISCUSSION THREAD

**Discussion by smokedzn** [Chatterino/chatterino2#4235] [Ideas] [answered]
I think there's big missed potential with using custom commands, especially if there would be more placeholders.

**As seen on [Commands - Chatterino](https://wiki.chatterino.com/Commands/) there _currently_ are the following placeholders:**

> **{1}, {2}, {3}** and so on can be used to insert the 1st, 2nd, 3rd, ... word after the trigger.
> **{1+}** can be used to insert all words starting with the 1st, ... word.
> **{channel.name}** can be used to insert the name of the current channel.
> **{channel.id}** can be used to insert the Twitch account ID of the owner of current channel. 
> **{my.name}** can be used to insert the Twitch username of the currently selected account. 
> **{my.id}** can be used to insert the Twitch account ID of the currently selected account. 
> **{stream.title}** can be used to insert the title set in the current channel. Only works if the channel is live.
> **{stream.game}** can be used to insert the game set in the current channel. Only works if the channel is live.

My suggestion, or idea in general is to add more placeholders which are pretty much missing in my opinion. 
Examples include:

> **{viewer.count}** can be used to list the amount of viewers watching the current channel. Only works if the channel is live.
> Example: Currently **{viewer.count}** viewers are watching.
> Example Output: Currently **5,914** viewers are watching.

> **{chatter.count}** can be used to list the amount of chatters in the current channel.
> Example: Currently **{chatter.count}** chatters are in chat.
> Example Output: Currently **5,693** chatters are in chat.

> **{previous.message}** can be used to insert the latest message from chat into your message.
> Example: He really said "**{previous.message}**"
> Example Output: He really said "**I'm going to gift 100 subs if you eat this DansGame**"

> **{previous.message.username}** can be used to insert the username from the latest message from chat into your message.
> Example: **{previous.message.username}** really do be saying weird things again.
> Example Output: **smokedzn1** really do be saying weird things again.

> **{username.random}** can be used to randomly pick a username from the current channel.
> Example: 🤜🔔 @**{username.random}** :tf:
> Example Output: 🤜🔔 @**SantaMouse** :tf:

> **{repeat.word.[amount]}** can be used to repeat the most written word (including emotes) from the last 10 to 100 messages.
> Example: **{repeat.word.25}**
> Example Output: **OMEGALUL**

> **{time}** can be used to insert your current default time. 
> Variables ( **{time.[variable]}** ) can be used: `h.mm | hh.mm | h.mm.a | hh.mm.a | h.mm.ss | hh.mm.ss | h.mm.ss.a | hh.mm.ss.a | h.mm.ss.zzz | h.mm.ss.zzz.a | hh.mm.ss.zzz | hh.mm.ss.zzz.a`
> Example: It's currently **{time.h.mm.ss.a}** for me
> Example Output: It's currently **12:22:58 am** for me

If you have any other ideas for placeholders please feel free to contribute to this discussion/idea, I'd really appreciate it and hope that some of the mentioned placeholders will be implemented in future versions.
Have a good day, everyone!

**Comment by pajlada** [comment, marked as answer]
:+1: `{viewer.count}` would be `{stream.viewer_count}` and should be easy to implement, we have the data already
:+1: `{chatter.count}` would be `{channel.num_chatters}` and should also be relatively easy to implement, some users will have this information but not all of them.
:+1: `{previous...}` would need to be a bit more explicit in what it's fetching, is it using context from the last message in a split? If so, we'd probably want `{split.previous_message.username}` or `{split.previous_message.text}`
:+1: `{time}` is interesting, bit more on the complex side but I could see some useful scenarios. Would need to be careful for how we handle the text formatting

:-1: `{username.random}` feels a bit overkill/abusey for my taste atm, unless someone has a proper use case for it.
:-1: `{repeat.word.[amount]}` doesn't have a super clear description, so it would be hard to correctly implement/describe to users so I feel like this doesn't fit.

Thanks for the detailed discussion, happy to hear other people's feedback on these additions

**Reply by smokedzn** [reply]
Thanks for the response and feedback!

`{username.random}` would be indeed rather used for jokes as random pinging, but I was also thinking about using it for chatting with random people before for example gifting them or generally raffles, altough I'm pretty sure there are especially for streamers easier ways using StreamElements or else, or just randomly right clicking a name, so you're definetly right. It would rather be abused than used as intended.

`{repeat.word.[amount]}` would be an easy way to react with chat when they are typing the same thing over and over without having to copy or type out the whole message and/or emote, as in some chats emotes and text are spammed over and over in specific moments _which is also fine with some of the streamers I have in mind_, but I don't think it would have any general value besides using it as a command for lazy people who can't hit Ctrl + C and Ctrl + V or typing 3 letters and pressing Tab. Would totally understand if this won't be added.

I'm pretty tired and worn out at the moment, so maybe I'll comment again with some other new ideas if I don't forget about it in the next few days, if that would be appreciated. 
**Maybe some other users also have some cool ideas, so feel free to comment on the discussion too!**

Would be really happy to see some of the mentioned commands implemented as the existing ones already really are useful, but still hold much potential for more interesting commands.
Keep up the good work!

- smokedzn

**Reply by Chomchaum** [reply]
I'd suggest another placeholder :

> {selected_message.username} would return the message username for commands triggered through message menu
> Example: Hello {selected_message.username}
> Example Output used on a message sent by Foo: Hello Foo
